launch_ros/launch_ros/actions/composable_node_container.py

```diff
@@ -77,7 +77,13 @@ def __init__(
                 )
             namespace = node_namespace
 
-        if not namespace:
+        if name is None:
+            raise RuntimeError("'name' is a required argument")
+
+        if namespace is None:
+            raise RuntimeError("'namespace' is a required argument")
+
+        if namespace == '':
```

Comment:
Just out of curiosity, what happens if `name == ''`?

Reply:
I think that that fails, because `Node` action is checking the validity of the name before launching the process.

The semantics of `namespace` is a bit difference:
- A relative `namespace` in launch means that it will be affected by previous `PushRosNamespace` actions.
- An absolute one no.

`Node` will evaluate an actual namespace based on the previous `PushRosNamespace` and the user provided argument, and then validate.
In Foxy `namespace=''` was behaving like no passing a remapping rule, thus that extra check is needed.

Reply:
Makes sense, thanks.